state/execution.go

```diff
@@ -84,6 +84,11 @@ func (blockExec *BlockExecutor) ApplyBlock(state State, blockID types.BlockID, b
 	// Save the results before we commit.
 	saveABCIResponses(blockExec.db, block.Height, abciResponses)
 
+	// Delete the results for the previous height (not needed anymore).
+	if block.Height > 0 {
+		deleteABCIResponses(blockExec.db, block.Height-1)
+	}
```

Comment:
Is better to change calcABCIResponsesKey return a constant string? as the State's key stateKey:  
stateKey = []byte("stateKey")

this may reduce some IO.

Reply:
Good idea